### PR TITLE
User/ProjectDB refactoring

### PR DESCRIPTION
**Pull request by michaelmattig**
Added project and view to session and refactored the database schema

### CODE REVIEW

File: services/src/projects/project.rs [hunk 1]
```diff
@@ -78,14 +72,16 @@ impl Project {
     }
 }
 
-#[derive(Debug, PartialEq, Serialize, Deserialize, Clone)]
+#[derive(Debug, PartialEq, Serialize, Deserialize, Clone, ToSql, FromSql)]
 pub struct STRectangle {
+    pub spatial_reference: SpatialReference,
```
Comment (ChristianBeilschmidt) [inline services/src/projects/project.rs:77]: Maybe `SpatialReferenceOption` to allow unreferenced rectangles

File: services/src/users/hashmap_userdb.rs [hunk 2]
```diff
@@ -39,7 +41,14 @@ impl UserDB for HashMapUserDB {
     async fn login(&mut self, user_credentials: UserCredentials) -> Result<Session> {
         match self.users.get(&user_credentials.email) {
             Some(user) if bcrypt::verify(user_credentials.password, &user.password_hash) => {
-                let session = Session::new(user);
+                let session = Session {
+                    id: SessionId::new(),
+                    user: user.id,
+                    created: chrono::Utc::now(),
+                    valid_until: chrono::Utc::now() + chrono::Duration::minutes(60),
```
Comment (ChristianBeilschmidt) [inline services/src/users/hashmap_userdb.rs:48]:
```suggestion
                    // TODO: make session length configurable
                    valid_until: chrono::Utc::now() + chrono::Duration::minutes(60),
```

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: services/src/projects/project.rs (hunk 1 above)
Comment (michaelmattig) [reply services/src/projects/project.rs:77]: we have `SpatialReference::Unreferenced` now

Comment (ChristianBeilschmidt) [reply services/src/projects/project.rs:77]: Yes, but this is in your branch still `SpatialReferenceOption::None`.

File: services/src/projects/project.rs [hunk 3]
```diff
@@ -75,14 +75,14 @@ impl Project {
...
-    pub spatial_reference: SpatialReference,
+    pub spatial_reference: SpatialReferenceOption,
     pub bounding_box: BoundingBox2D,
     pub time_interval: TimeInterval,
 }
 
 impl STRectangle {
     pub fn new<A, B>(
-        spatial_reference: SpatialReference,
+        spatial_reference: SpatialReferenceOption,
```
Comment (ChristianBeilschmidt) [inline services/src/projects/project.rs:85]: You already say `time_start: A, ` … `where A: Into<TimeInstance>`,
you could be consistent and say `spatial_reference: S` … `,where S: Into<SpatialReferenceOption>`.
Then you can omit writing `wgs84().into()`.

Comment (michaelmattig) [reply services/src/projects/project.rs]: 8966c1c9b1bff565fcdc74fd20f8d6b5cc41ddb6

**Pull request merged**